tests/core/netplan-cfg/task.yaml

```diff
@@ -26,21 +26,23 @@ execute: |
     snap get -d system system.network.netplan | jq .
 
     echo "Check that setting adding a br54 interface via netplan works"
-    snap set system system.network.netplan.network.bridges.br54.dhcp4=true
+    # set to false avoids that networkd waits for a valid dhcp record
+    # here (which will never come), see LP:1967084 for details
+    snap set system system.network.netplan.network.bridges.br54.dhcp4=false
 
     echo "Check that the interface is really there"
     netplan get | MATCH br54
     ip link | MATCH br54
     res=$(yq e '.network.bridges.br54.dhcp4' - < /etc/netplan/90-snapd-config.yaml)
-    [ "$res" = "true" ]
+    [ "$res" = "false" ]
```

Comment:
Since this is now testing a negative, I wonder if it can be replaced by some other check for the existance of `.network.bridges.br54`, maybe something like (untested!):

    res=$(yq e '.network.bridges.br54 | length' - < /etc/netplan/90-snapd-config.yaml)
    [ "$res" -gt 0 ]

or maybe even

    # if there is some other key besides dhcp4, better use the other one!
    res=$(yq e '.network.bridges.br54 | has(dhcp4)' - < /etc/netplan/90-snapd-config.yaml)
    [ "$res" -gt 0 ]

Your current check is correct, but it just looks a bit weird here.

Reply:
Indeed, it looks a bit strange. I updated the comments to explain what is going on a bit better.